opengrok-indexer/src/main/java/org/opengrok/indexer/util/CtagsUtil.java

```diff
@@ -86,6 +91,43 @@ public static boolean validate(String ctagsBinary) {
         return result;
     }
 
+    /**
+     * Deletes ctags temporary files left over after terminating ctags processes
+     * in case of timeout, {@see Ctags#doCtags}.
+     */
+    public static void deleteTempFiles() {
+        String[] dirs = {System.getProperty("java.io.tmpdir"),
+                System.getenv("TMPDIR"), System.getenv("TMP")};
+
+        for (String dir : dirs) {
+            deleteTempFiles(dir);
+        }
+    }
+
+    private static void deleteTempFiles(String directoryName) {
+        final Pattern pattern = Pattern.compile("tags\\.\\S{6}"); // ctags uses this pattern to call mkstemp()
+
+        if (directoryName == null) {
+            return;
+        }
+
+        File dir = new File(directoryName);
+        File[] files = dir.listFiles((dir1, name) -> {
+            Matcher matcher = pattern.matcher(name);
+            return matcher.find();
+        });
+
+        for (File file : files) {
+            if (file.isFile()) {
```

Comment:
How about simpler just as `if (!file.delete()) {` and just the log statement inside with no exception handling?

Reply:
that's nicer, changed.